src/DonationForms/DataTransferObjects/DonateFormRouteData.php

```diff
@@ -83,8 +85,31 @@ public function validated(): DonateControllerData
             $this->throwDonationFormFieldErrorsException($validator->errors());
         }
 
+        $validData->billingAddress = new BillingAddress();
+
         foreach ($validator->validated() as $fieldId => $value) {
-            $validData->{$fieldId} = $value;
+            switch ($fieldId) {
+                case 'country':
+                    $validData->billingAddress->country = $value;
+                    break;
+                case 'address1':
+                    $validData->billingAddress->address1 = $value;
+                    break;
+                case 'address2':
+                    $validData->billingAddress->address2 = $value;
+                    break;
+                case 'city':
+                    $validData->billingAddress->city = $value;
+                    break;
+                case 'state':
+                    $validData->billingAddress->state = $value;
+                    break;
+                case 'zip':
+                    $validData->billingAddress->zip = $value;
+                    break;
+                default:
+                    $validData->{$fieldId} = $value;
```

Comment:
Ah I see why you had to do this.  It definitely works but adds additional granular complexity to this validation piece that I would like to avoid if possible.  

2 alternative ideas.
1. We assign all the billing address properties to the controller data DTO individually and add a new method called `getBillingAddress()` that builds the object when called.
2. We come up with a clever dot notion field name syntax that tries to auto-assign object properties.  Example:

```
$fieldName = 'billingAddress.country';
$field = explode('.', $fieldName);
$validData->{$field[0]}->{$field[1]} = $value;
```

I'm leaning toward option 1 since it's the simplest.  Open to your thoughts.

Reply:
I liked the first option. Resolved in ec6cfbc54d3bb80fa7bf80c462305417b27a71d3